support-frontend/app/services/pricing/DefaultPromotionsService.scala

```diff
@@ -0,0 +1,69 @@
+package services.pricing
+
+import com.amazonaws.services.s3.AmazonS3URI
+import com.gu.aws.AwsS3Client
+import com.gu.support.config.{Stage, TouchPointEnvironments}
+import io.circe._
+import io.circe.parser._
+import io.circe.generic.auto._
+import akka.actor.ActorSystem
+import com.gu.support.catalog.{GuardianWeekly, Product}
+import com.typesafe.scalalogging.LazyLogging
+import services.pricing.DefaultPromotionsService.DefaultPromotions
+
+import java.util.concurrent.atomic.AtomicReference
+import scala.concurrent.ExecutionContext
+import scala.concurrent.duration.DurationInt
+import scala.util.Try
+
+trait DefaultPromotionsService {
+  def getPromos(product: Product): List[String]
+}
+
+object DefaultPromotionsService {
+  case class DefaultPromotions(guardianWeekly: List[String])
+
+  implicit val decoder = Decoder[DefaultPromotions]
+}
+
+class DefaultPromotionsServiceS3(
+    client: AwsS3Client,
+    stage: Stage,
+    system: ActorSystem,
+)(implicit ec: ExecutionContext)
+    extends DefaultPromotionsService
+    with LazyLogging {
+
+  private val s3Uri = {
+    val env = TouchPointEnvironments.fromStage(stage)
+    new AmazonS3URI(s"s3://gu-promotions-tool-private/${env.envValue}/defaultPromos.json")
+  }
+  private val defaultPromos = new AtomicReference[DefaultPromotions](
+    DefaultPromotions(guardianWeekly = Nil),
+  )
+
+  private def fetch(): Try[DefaultPromotions] =
+    for {
+      raw <- client.fetchAsString(s3Uri)
+      defaultPromos <- decode[DefaultPromotions](raw).toTry
+    } yield defaultPromos
+
+  private def update(): Try[Unit] = fetch().map(newPromos => defaultPromos.set(newPromos))
+
+  private def startPollingS3(): Unit =
+    system.scheduler.scheduleWithFixedDelay(0.minutes, 1.minute) { () =>
```

Comment:
I wonder if this really needs to poll every minute, these change very infrequently. I feel like we could probably push it up to 10 mins or more and just let make it clear in the tooling that it will take a while?

Reply:
interesting thought - do you think it may be not necessary on cost reasons, or reliability, clarity? or something else?

Reply:
I guess just on the principal of not doing unnecessary work. It feels wasteful to be polling every minute for something which can remain unchanged for months at a time.

Reply:
fair enough, as long we keep the time taken to update smaller than the time for a bug report to be made, we will be ok!